destination/writer/writer.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	return nil
+}
+
+// Update updates a record.
+func (w *Writer) Update(ctx context.Context, record sdk.Record) error {
+	tableName := w.getTableName(record.Metadata)
+
+	payload, err := w.structurizeData(record.Payload.After)
+	if err != nil {
+		return fmt.Errorf("structurize payload: %w", err)
+	}
+
+	// if payload is empty return empty payload error
+	if payload == nil {
+		return errNoPayload
+	}
+
+	payload, err = columntypes.ConvertStructureData(w.columnTypes, payload)
+	if err != nil {
+		return fmt.Errorf("convert structure data: %w", err)
+	}
+
+	key, err := w.structurizeData(record.Key)
+	if err != nil {
+		return fmt.Errorf("structurize key: %w", err)
+	}
+
+	keyColumns, err := w.getKeyColumns(key)
+	if err != nil {
+		return fmt.Errorf("get key columns: %w", err)
+	}
+
+	// remove keys from the payload
+	for i := range keyColumns {
+		delete(payload, keyColumns[i])
+	}
+
+	columns, values := w.extractColumnsAndValues(payload)
+
+	updateColumns := make([]string, len(columns))
+	for i := range columns {
+		updateColumns[i] = fmt.Sprintf("%s = %s", columns[i], placeholder)
+	}
+
+	whereClauses := make([]string, len(keyColumns))
+	for i := range keyColumns {
+		whereClauses[i] = fmt.Sprintf("%s = %s", keyColumns[i], placeholder)
+		values = append(values, key[keyColumns[i]])
+	}
+
+	query := fmt.Sprintf(updateFmt, tableName, strings.Join(updateColumns, coma), strings.Join(whereClauses, operatorAnd))
+
+	_, err = w.db.ExecContext(ctx, query, values...)
+	if err != nil {
+		return fmt.Errorf("exec update %q, %v: %w", query, values, err)
+	}
+
+	return nil
+}
+
+// Delete deletes a record.
+func (w *Writer) Delete(ctx context.Context, record sdk.Record) error {
+	tableName := w.getTableName(record.Metadata)
+
+	key, err := w.structurizeData(record.Key)
+	if err != nil {
+		return fmt.Errorf("structurize key: %w", err)
+	}
+
+	keyColumns, err := w.getKeyColumns(key)
+	if err != nil {
+		return fmt.Errorf("get key columns: %w", err)
+	}
+
+	whereClauses := make([]string, len(keyColumns))
+	values := make([]any, 0, len(keyColumns))
+	for i := range keyColumns {
+		whereClauses[i] = fmt.Sprintf("%s = %s", keyColumns[i], placeholder)
+		values = append(values, key[keyColumns[i]])
+	}
+
+	query := fmt.Sprintf(deleteFmt, tableName, strings.Join(whereClauses, operatorAnd))
+
+	_, err = w.db.ExecContext(ctx, query, values...)
+	if err != nil {
+		return fmt.Errorf("exec delete %q, %v: %w", query, values, err)
+	}
+
+	return nil
+}
+
+// returns either the record metadata value for the table
+// or the default configured value for the table.
+func (w *Writer) getTableName(metadata map[string]string) string {
+	tableName, ok := metadata[metadataTable]
+	if !ok {
+		return w.table
+	}
+
+	return tableName
+}
+
+// returns either all the keys of the sdk.Record's Key field.
+func (w *Writer) getKeyColumns(key sdk.StructuredData) ([]string, error) {
+	if len(key) == 0 {
+		return nil, errEmptyKey
+	}
```

Comment:
We should have a fallback (e.g. use columns from the payload).

Reply:
I'm sorry, I didn't catch you.
I've added this method to get the column names from a key in order to put them in the where clause in SQL queries.
Why do we need to get it from the payload?

Reply:
My point was: the key may be missing, but the payload will likely contain the key, because it will be the row. We can have the destination key column configurable, and then extract it from the payload.

Reply:
So, we've decided to add the additional configurable parameter `keyColumn` and try to find the value in the `record.Payload` to use it as a `record.Key`.

Reply:
I've added this logic.